source/uwp/SharedRenderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -356,14 +357,23 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering
 
         if (schemeName == L"data")
         {
-            // Decode base 64 string
-            winrt::hstring dataPath = imageUrl.Path();
-            std::string data = AdaptiveBase64Util::ExtractDataFromUri(HStringToUTF8(dataPath));
-            std::vector<char> decodedData = AdaptiveBase64Util::Decode(data);
-
             winrt::DataWriter dataWriter{winrt::InMemoryRandomAccessStream{}};
 
-            dataWriter.WriteBytes(std::vector<byte>{decodedData.begin(), decodedData.end()});
+            auto foundBase64 = HStringToUTF8(imageUrl.Path()).find("base64");
```

Comment:
Can you first check `if (imgProperties.isImageSvg)` and, only if that fails, proceed with decoding the base64 string? In this way, we don’t call HstringToUTF8 unnecessarily.

Also, can you cache the result of `HstringToUTF8(imageUrl.Path())` as dataPath, and reuse it for the foundBase64 check and data extraction, since they are the same string?

Reply:
The image can be base64 and svg. It can also be bitmap and base64. We follow the same code path for base64. If that is not a match and the image was svg, we try the svg xml code path.

Reply:
if so, please disregard the first paragraph.